src/Storages/StorageURL.cpp

```diff
@@ -185,11 +186,13 @@ namespace
                         std::string user_info = request_uri.getUserInfo();
                         if (!user_info.empty())
                         {
-                            std::size_t n = user_info.find(':');
+                            std::string url_decoded_user_info;
+                            Poco::URI::decode(user_info, url_decoded_user_info);
+                            std::size_t n = url_decoded_user_info.find(':');
                             if (n != std::string::npos)
                             {
-                                credentials.setUsername(user_info.substr(0, n));
-                                credentials.setPassword(user_info.substr(n + 1));
+                                credentials.setUsername(url_decoded_user_info.substr(0, n));
+                                credentials.setPassword(url_decoded_user_info.substr(n + 1));
```

Comment:
Why not also set just username if there is no password like here https://github.com/ClickHouse/ClickHouse/pull/33299/files#r780763660 ?